src/lib.rs

```diff
@@ -68,6 +113,113 @@ pub struct PortGroup {
 pub struct Port {
     // Path of the domain socket.
     pub socket_path: &'static str,
+    pub socket: UnixStream,
+}
+
+impl Port {
+    pub fn new(path: &'static str) -> Port {
+        // Connect to the unix domain socket for this port
+        let socket = UnixStream::connect(path).unwrap();
+        // Create and return the port struct
+        Port {
+            socket_path: path,
+            socket: socket,
+        }
+    }
+
+    pub fn i2c(&mut self, address: u8, frequency: u32) -> I2C {
+        // Create and return the I2C struct
+        I2C::new(self, address, frequency)
```

Comment:
Thanks for the tips, Kevin. I didn't notice that particular nuance of this API until you brought it up. My instinct is to go with the first option so there is less cognitive load on the API user. Do you think one of those two options is more intuitive for a Rust user?